front/pages/api/w/[wId]/data_sources/managed.ts

```diff
@@ -332,24 +332,27 @@ async function handler(
         ? VaultResource.fetchWorkspaceGlobalVault(auth)
         : VaultResource.fetchWorkspaceSystemVault(auth));
       const dataSource = await DataSourceResource.makeNew({
-        name: dataSourceName,
+        assistantDefaultSelected,
+        connectorProvider: provider,
         description: dataSourceDescription,
         dustAPIProjectId: dustProject.value.project.project_id.toString(),
-        workspaceId: owner.id,
-        assistantDefaultSelected,
         editedByUserId: user.id,
+        name: dataSourceName,
         vaultId: vault.id,
+        workspaceId: owner.id,
       });
 
-      const globalVault = vault.isGlobal()
-        ? vault
-        : await VaultResource.fetchWorkspaceGlobalVault(auth);
-
       // For managed data source, we create a default view in the workspace vault.
-      await DataSourceViewResource.createViewInVaultFromDataSourceIncludingAllDocuments(
-        globalVault,
-        dataSource.toJSON()
-      );
+      if (dataSource.isManaged()) {
+        const globalVault = vault.isGlobal()
+          ? vault
+          : await VaultResource.fetchWorkspaceGlobalVault(auth);
```

Comment:
if dataSource is managed, vault is always system , you can remove the test and always fetch global